impl/src/settings.rs

```diff
@@ -370,7 +381,35 @@ pub fn load_settings<T: AsRef<Path>>(cargo_toml_path: T) -> Result<RazeSettings,
     return Err(RazeError::Generic(err.to_string()));
   }
   let mut settings = match toml::from_str::<CargoToml>(&toml_contents) {
```

Comment:
Do you know if `package.metadata.raze` settings can be parsed from `cargo metadata`?

Specifically, could something like the `CargoMetadataFetcher` be used to load these settings more holistically?

Reply:
Indeed, they can. It looks like this:

```shell
$ cargo metadata --format-version 1 | jq '.. | .raze? | select(.)'
{
  "genmode": "Remote",
  "workspace_path": "//third_party/rust",
  "crates": {
    "crc": {
      "1.8.1": {
        "gen_buildrs": true
      }
    },
    "indexmap": {
      "1.6.0": {
        "additional_flags": [
          "--cfg=has_std"
        ]
      }
    },
    "proc-macro2": {
      "1.0.24": {
        "additional_flags": [
          "--cfg=use_proc_macro",
          "--cfg=wrap_proc_macro"
        ]
      }
    },
    "prost-build": {
      "0.6.1": {
        "gen_buildrs": true
      }
    },
    "thiserror": {
      "1.0.21": {
        "gen_buildrs": true
      }
    }
  }
}
```

The path is `.packages[].metadata.raze`, so you’d have to pluck it from
the right package.

I haven’t looked at `CargoMetadataFetcher`. If you’d prefer that
approach, I can do so (but no promises on when I’ll get around to it).

Note that `[raze]` does not show up in `cargo metadata`, so as long as
you want to keep both around you would need two parsing paths. So
I might propose a plan of (a) parse both from `Cargo.toml`, (b) wait
until comfortable dropping compat, (c) remove `CargoToml::raze`,
(d) migrate to parsing from `cargo metadata` now that that’s easier.

Reply:
I was more trying to find out if this information could be parsed all from `cargo metadata`.

Reply:
Looks like the `[package.metadata.raze]` info can; the `[raze]` info can’t.